ns/ref.go

```diff
@@ -0,0 +1,280 @@
+// Package ns defines the qri dataset naming system
+// This package is currently an experiment while we try to settle the distinction
+// between a dataset.datasetPod and a reference. This code is extracted from
+//  github.com/qri-io/qri/repo/ref.go
+//
+// This package will either move to a new location or be removed entirely
+// before 2019-04-01
+//
+// initial RFC is here: https://github.com/qri-io/rfcs/blob/master/text/0006-dataset_naming.md
+package ns
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/mr-tron/base58"
+	multihash "github.com/multiformats/go-multihash"
+)
+
+var (
+	// ErrEmptyRef indicates that the given reference is empty
+	ErrEmptyRef = fmt.Errorf("repo: empty dataset reference")
+)
+
+// Ref encapsulates a reference to a dataset. This needs to exist to bind
+// ways of referring to a dataset to a dataset itself, as datasets can't easily
+// contain their own hash information, and names are unique on a per-repository
+// basis.
```

Comment:
I think the rest of this description is unnecessary, but if you want to really drive the point home, say that "Ref" is  analogous to a FilePath, not a FileHandle, and doesn't contain any of the contents of a dataset, only the name used to retrieve it.